htdp-lib/htdp/bsl/reader.rkt

```diff
@@ -26,7 +26,7 @@
 (define ((make-info options) key default use-default)
   (case key
     [(drscheme:toolbar-buttons)
-     (list (dynamic-require 'stepper/drracket-button 'stepper-drracket-button)
+     (list ((dynamic-require 'lang/private/sl-stepper-button 'sl-stepper-drracket-button) options)
            (dynamic-require 'drracket/syncheck-drracket-button 'syncheck-drracket-button))]
 
     [(drscheme:opt-out-toolbar-buttons)
```

Comment:
Okay, it seems weird that there's asymmetry in that we're only changing bsl/reader.rkt and not bsl+/reader.rkt etc. However, after some digging, it looks like *all* the languages refer to bsl/reader.rkt (or, more specifically, the files such as isl/lang/reader.rkt specify this file as as their language in a #lang spec. 

It looks to me like this file probably shouldn't live here. I also have no appetite for fixing this problem, especially right before the release. After the release would it be reasonable to add a comment such as this?

;; this file defines the #lang language used to specify all of the teaching languages. It's not bsl-specific.

Reply:
Which file shouldn't live here? The stepper file should live in the stepper collection you mean?

Reply:
No; it appeared to me that this file (bsl/reader.rkt) is not specific to bsl; instead, it appears to define a language that's used as a #lang by *all* of the teaching language langs: bsl/lang/reader.rkt, bsl+/lang/reader.rkt, isl/lang/reader.rkt, etc. 

Here's what I mean:

```
hardy:~/htdp/htdp-lib/htdp (git)-[master]- clements> head -1 bsl/lang/reader.rkt
#lang s-exp htdp/bsl/reader
hardy:~/htdp/htdp-lib/htdp (git)-[master]- clements> head -1 bsl+/lang/reader.rkt
#lang s-exp htdp/bsl/reader
hardy:~/htdp/htdp-lib/htdp (git)-[master]- clements> head -1 isl/lang/reader.rkt
#lang s-exp htdp/bsl/reader
hardy:~/htdp/htdp-lib/htdp (git)-[master]- clements> head -1 isl+/lang/reader.rkt
#lang s-exp htdp/bsl/reader
hardy:~/htdp/htdp-lib/htdp (git)-[master]- clements> head -1 asl/lang/reader.rkt
#lang s-exp htdp/bsl/reader
```
In other words, it appears that this file is a general basis for defining student languages, not something that's specific to bsl.

Really, this isn't a big deal; I'm just reading through all this code and trying to figure out what it means.

Reply:
I agree with @jbclements - this confused the heck out of me, too.

(In the DeinProgramm languages, the corresponding file lives in sdp/private/reader.rkt, not in the directory of a specific language level.)